torrent.go

```diff
@@ -122,6 +123,44 @@ func (t *Torrent) Closed() <-chan struct{} {
 	return t.closed.LockedChan(&t.cl.mu)
 }
 
+// KnownSwarm returns the known subset of the peers in the Torrent's swarm, including active,
+// pending, and half-open peers.
+func (t *Torrent) KnownSwarm() (ks []Peer) {
+	// Add pending peers to the list
+	for _, peer := range t.peers {
+		ks = append(ks, peer)
+	}
+
+	// Add active peers to the list
+	for conn := range t.conns {
+		host, portString, err := net.SplitHostPort(conn.remoteAddr().String())
+		if err != nil {
+			panic(err)
+		}
+
+		ip := net.ParseIP(host)
+		port, err := strconv.Atoi(portString)
+		if err != nil {
+			panic(err)
+		}
+
+		ks = append(ks, Peer{
+			Id: conn.PeerID,
+			IP: ip,
+			Port: port,
+			Source: conn.Discovery,
+			// TODO: the connection can be unencrypted due to our (or the peer's) preference,
```

Comment:
SupportsEncryption is advisory only. I don't believe anything actually acts on it, but the information is provided by some sources.

Reply:
> but the information is provided by some sources

In this case, by which sources? `conn.encrypted` is enough or do I have another option?

Reply:
If the connection is encrypted, that's certainly enough to set SupportsEncryption. But if we're not connected to them with an encrypted connection, I couldn't say what's appropriate. We can carry forward the SupportsEncryption value as we received it from trackers/DHT/PEX, or just use the encryption state for the connection. It's probably easiest to do the latter for now.